dataops/gcp/gcp_setup_template.yml

```diff
@@ -1,12 +1,28 @@
 variables:
+  GCP_KEY_PATH: TBS
+  GCP_DOCKER_KEY_PATH: TBS
+  GIT_KEY_PATH: TBD
   DOCKER_REGISTRY: europe-central2-docker.pkg.dev
   BLOB_CONFIG_PATH: blob_args.json
 
-.setup:
+.gcp_setup: &gcp_setup
```

Comment:
It would neat to use workload identity federation rather than credentials.json from SA, I think it can be more secure and have smaller overhead because we don't need to generate the file and put it in variables for CICD

Some references:
https://github.com/Cyclenerd/google-workload-identity-federation#set-up-identity-federation-for-gitlab-ci
https://github.com/Cyclenerd/google-workload-identity-federation/blob/master/.gitlab-ci.yml
https://cloud-architecture-design.medium.com/using-gitlab-ci-with-gcp-workload-identity-federation-8b80bec99c0b

Reply:
but to execute gcloud iam workload-identity-pools you still need SA key isn't it? So what is the difference?

Reply:
No, we are generating this key using jwt token taken from job env https://docs.gitlab.com/ee/ci/cloud_services/index.html#how-it-works
And we are using this token as source and pointing to specific workload identity we can generate temporal credentials for specific job

Reply:
But I think implementing this right now would be big overhead (changes in terraform), maybe a feature request?